### DISCUSSION TITLE
If i want Each browser is different Chromium object use CustomeScheme Object,some Unexpected question

### DISCUSSION THREAD

**Discussion by PingXD** [cefsharp/CefSharp#3946] [General]
Each browser custom protocol is the same, but use different IResourceSchemeHandlerFactory copy
The implementation is as follows:

>          var setting = new RequestContextSettings();
            RequestContext req = new RequestContext(setting);
            var scheme = new HttpBridgeSchemeHandle(chromiumWebBrowser, this);  //this object use ISchemeHandlerFactory interface
            req.RegisterSchemeHandlerFactory("https", "bridge.com", scheme);
            chromiumWebBrowser.RequestContext = req;

I use the same custom protocol for each browser instance,But the constructor is the browser object currently created
That is, each browser object has its own SchemeHandlerFactory
However, each time a new browser object is created, the existing browser object ISchemeHandlerFactory scheme will be replaced by a new one
It's like creating A browser A with A1 SchemeHandlerFactory and then creating A browser B with B1 SchemeHandlerFactory where browser A would want to use A1Factory but now browser A is using B1Factory

**Comment by amaitland** [comment]
What's your question exactly?

**Reply by PingXD** [reply]
Each old ISchemeHandlerFactory scheme will be replaced by a new one
Creating A browser as A use A1 SchemeHandlerFactory and then creating  browser B with B1 SchemeHandlerFactory where browser A would want to use A1Factory but now browser A is using B1Factory
**Question:**
How do I isolate each browser's SchemeHandlerFactory from each other

**Reply by amaitland** [reply]
> How do I isolate each browser's SchemeHandlerFactory from each other

They are tied to a RequestContext. A RequestContext van be shared by one or more ChromiumWebBrowser instances. 

Still not quite sure I understand what you are asking.

**Reply by PingXD** [reply]
ok,The problem has been solved
now has a new problem: I use RequestContext to register SchemeHandler how  allow CROS, my custome scheme is 'xbro', It is not a standard protocol
`            RequestContextSettings rcs = new RequestContextSettings();
            rcs.CachePath = Program.TempPath();
            RequestContext req = new RequestContext(rcs, new RequestContextHandle(this, chromiumWebBrowser));
            req.RegisterSchemeHandlerFactory("xbro", null, new ResourceSchemeHandlerFactory());
            chromiumWebBrowser.RequestContext = req;`
![image](https://user-images.githubusercontent.com/48038690/148331685-e213bcc8-f962-44e3-9976-b6f167198972.png)
![image](https://user-images.githubusercontent.com/48038690/148331709-c799c927-e435-4cb7-9ef7-af4348fdda00.png)

**Reply by amaitland** [reply]
> allow CROS, my custome scheme is 'xbro', It is not a standard protocol

Making CORS requests to a custom scheme is not currently supported. 

- Use https scheme and specify a domain when registering scheme handler factory. 
- intercept arbitrary requests using http://cefsharp.github.io/api/96.0.x/html/M_CefSharp_IRequestContextHandler_GetResourceRequestHandler.htm
- intercept arbitrary requests using https://github.com/cefsharp/CefSharp/wiki/General-Usage#request-interception
